### PR TITLE
[WIP] Calibration plugin

### PR DESCRIPTION
**Pull request by ayush-1506**
`filter_asr_output` is the main filtering function here. Model loading will be done at the template side. The functions expects `List[List[Dict[str, Any]]` as input.

On the template side, we need to load the model and call this just before normalize. Another PR for that coming soon.

### CODE REVIEW

File: dialogy/plugins/preprocess/text/calibration.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
+from typing import Any, List, Dict
+
+import math
+
+
+def predict_alternative(
+    alternative: Dict[str, Any], vectorizer: Any, classifier: Any
+) -> float:
+    vec_ = vectorizer.transform([alternative["transcript"]]).tolist()[0]
```
Comment (ltbringer) [inline dialogy/plugins/preprocess/text/calibration.py:9]: let's split just once and use it?
[This](https://github.com/Vernacular-ai/dialogy/blob/2f688ad30ed14f220eb8ae1a7b82fbe04ad2edbf/dialogy/constants/__init__.py#L17) prevents typos and provides autocomplete.

File: dialogy/plugins/preprocess/text/calibration.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+        1. :code:`List[List[Dict[str, Any]]]`
+
+    :type utterances: List[List[Dict[str, Any]]]
+    :return: Good alternatives.
+    :rtype: List[List[Dict[str, Any]]]
+    :raises: `AssertionError` if utterance isn't in the desired format.
+    """
+
+    assert len(utterances) == 1, f"Got {len(utterances)} instead of 1"
```
Comment (ltbringer) [inline dialogy/plugins/preprocess/text/calibration.py:57]: This (`len(utterance) > 1`) is valid for google asr. Might become true for our ASRs too.

**Review by ltbringer — COMMENTED** [review]
Coverage has fallen, can we mock the model and have some features tested?

File: pyproject.toml [hunk 3]
```diff
@@ -5,7 +5,7 @@ description = "Language understanding for human dialog."
 authors = ["Amresh Venugopal <amresh.venugopal@gmail.com>"]
 
 [tool.poetry.dependencies]
-python = "^3.7"
+python = "3.7"
```
Comment (ayush-1506) [inline pyproject.toml:8]: Had to change this because of some scipy dependency.

**Review by ltbringer — APPROVED** [review]

**Pull request merged**